src/Microsoft.OData.Core/Evaluation/LiteralFormatter.cs

```diff
@@ -233,6 +239,11 @@ private static string FormatRawLiteral(object value)
                 return value.ToString();
             }
 
+            if (value is TimeOnly timeOnly)
+            {
+                return ((TimeOfDay)timeOnly).ToString();

```

Comment:
Similar concerns as https://github.com/OData/odata.net/pull/3078/files#r1792817270 on whether we can avoid casting to `TimeOfDay` and handle `TimeOnly` directly.